packages/@aws-cdk/aws-appmesh/lib/mesh.ts

```diff
@@ -0,0 +1,313 @@
+    });
+  }
+
+  /**
+   * Exports the Mesh properties to re-use in other stacks
+   */
+  public abstract export(): MeshImportProps;
+}
+
+/**
+ * The set of properties used when creating a Mesh
+ */
+export interface MeshProps {
+  /**
+   * The name of the Mesh being defined
+   */
+  readonly meshName?: string;
+
+  /**
+   * The spec to be applied to the mesh, at this point, only egressFilter is available
+   */
+  readonly meshSpec?: MeshSpec;
+}
+
+/**
+ * Defines a new AppMesh mesh
+ *
+ * @see https://docs.aws.amazon.com/app-mesh/latest/userguide/meshes.html
+ */
+export class Mesh extends MeshBase {
+  /**
+   * A static method to import a mesh an make it re-usable accross stacks
+   */
+  public static import(scope: cdk.Construct, id: string, props: MeshImportProps): IMesh {
+    return new ImportedMesh(scope, id, props);
+  }
+
+  /**
+   * The name of the AppMesh mesh
+   */
+  public readonly meshName: string;
+
+  /**
+   * The Amazon Resource Name (ARN) of the AppMesh mesh
+   */
+  public readonly meshArn: string;
+
+  /**
+   * The unique identifier for the mesh
+   */
+  public readonly meshUid: string;
+
+  constructor(scope: cdk.Construct, id: string, props: MeshProps = {}) {
+    super(scope, id);
+
+    const name = props.meshName || this.node.id;
+
+    const filter = (props.meshSpec && props.meshSpec.egressFilter) || MeshFilterType.DROP_ALL;
+
+    const mesh = new CfnMesh(this, 'AppMesh', {
+      meshName: name,
+      spec: {
+        egressFilter: {
+          type: filter,
+        },
+      },
+    });
+
+    this.meshName = mesh.meshName;
+    this.meshArn = mesh.meshArn;
+    this.meshUid = mesh.meshUid;
+  }
+
+  /**
+   * Exports the Mesh properties to re-use in other stacks
+   */
+  public export(): MeshImportProps {
+    return {
+      meshName: new cdk.CfnOutput(this, 'MeshName', { value: this.meshName }).makeImportValue().toString(),
+      meshArn: new cdk.CfnOutput(this, 'MeshArn', { value: this.meshArn }).makeImportValue().toString(),
+      meshUid: new cdk.CfnOutput(this, 'MeshUid', { value: this.meshUid }).makeImportValue().toString(),
+    };
+  }
+}
+
+/**
+ * Use when import is called on Mesh.import(), returns properties that allows the mesh to be used
+ * accross stacks.
+ */
+export class ImportedMesh extends MeshBase {
+  /**
+   * The name of the AppMesh mesh
+   */
+  public readonly meshName: string;
+
+  /**
+   * The Amazon Resource Name (ARN) of the AppMesh mesh
+   */
+  public readonly meshArn: string;
+
+  /**
+   * The unique identifier for the mesh
+   */
+  public readonly meshUid: string;
+
+  constructor(scope: cdk.Construct, id: string, private readonly props: MeshImportProps) {
+    super(scope, id);
+
+    this.meshName = props.meshName;
+
+    this.meshArn =
+      props.meshArn ||
+      this.node.stack.formatArn({
+        service: 'appmesh',
+        resource: `mesh`,
+        resourceName: props.meshName,
+      });
+
+    this.meshUid = props.meshUid;
```

Comment:
This is what I'm talking about here in my comment, ARN is easy to generate, though the meshUid is not, unless I'm missing something, so making it option in an import call is not feasible. Even though the only real required property to find the mesh is the meshName.